packages/governance/src/contractGovernor.js

```diff
@@ -135,18 +136,73 @@ const start = async zcf => {
   };
 
   // CRUCIAL: only contractGovernor should get the ability to update params
-  /** @type {Promise<LimitedCreatorFacet>} */
   const limitedCreatorFacet = E(governedCF).getLimitedCreatorFacet();
-  const { voteOnParamChange, createdQuestion } = await setupGovernance(
-    zoe,
-    E(governedCF).getParamMgrRetriever(),
-    governedInstance,
-    timer,
-  );
+
+  let currentInvitation;
+  let poserFacet;
+  /** @type {() => Promise<PoserFacet>} */
+  const getUpdatedPoserFacet = async () => {
+    const newInvitation = await E(
+      E(E(governedCF).getParamMgrRetriever()).get({ key: 'main' }),
+    ).getInternalParamValue(CONTRACT_ELECTORATE);
+
+    if (newInvitation !== currentInvitation) {
+      poserFacet = E(E(zoe).offer(newInvitation)).getOfferResult();
+      currentInvitation = newInvitation;
+    }
+    return poserFacet;
+  };
+  await getUpdatedPoserFacet();
+  assert(poserFacet, X`question poser facet must be initialized`);
+
+  // All governed contracts have at least a governed electorate
+  const { voteOnParamChange, createdQuestion: createdParamQuestion } =
+    await setupParamGovernance(
+      zoe,
+      E(governedCF).getParamMgrRetriever(),
+      governedInstance,
+      timer,
+      getUpdatedPoserFacet,
+    );
+
+  // this conditional was extracted so both sides are equally asynchronous
+  /** @type {() => Promise<ApiGovernor>} */
+  const initApiGovernance = async () => {
+    // @ts-ignore `governedApis` is present on contracts wiht API invocation.
+
+    const [governedApis, governedNames] = await Promise.all([
+      E(governedCF).getGovernedApis(),
+      E(governedCF).getGovernedApiNames(),
+    ]);
+    if (governedNames.length) {
+      return setupApiGovernance(
+        zoe,
+        governedInstance,
+        governedApis,
+        governedNames,
+        timer,
+        getUpdatedPoserFacet,
+      );
+    }
+
+    // if we aren't governing APIs, voteOnApiInvocation shouldn't be called
+    return {
+      voteOnApiInvocation: () => {
+        throw Error('api governance not configured');
+      },
+      createdQuestion: () => false,
+    };
+  };
+
+  const { voteOnApiInvocation, createdQuestion } = await initApiGovernance();
```

Comment:
Above there's a helpful renaming destructure: `createdQuestion: createdParamQuestion`
```suggestion
  const { voteOnApiInvocation, createdQuestion: createdApiQuestion } = await initApiGovernance();
```

Reply:
good call.